web/src/debug.ts

```diff
@@ -10,42 +12,36 @@
     evaluates to foo() and prints the elapsed time
     to the console along with the name "foo". */
 
-export function print_elapsed_time(name, fun) {
+export function print_elapsed_time<T>(name: string, fun: () => T): T {
     const t0 = Date.now();
     const out = fun();
     const t1 = Date.now();
-    console.log(name + ": " + (t1 - t0) + " ms");
+    console.log(`${name} :  ${t1 - t0} ms`);
     return out;
 }
 
-export function check_duplicate_ids() {
-    const ids = new Set();
-    const collisions = [];
+export function check_duplicate_ids(): {collisions: Collision[]; total_collisions: number} {
+    const ids = new Set<string>();
+    const collisions: Collision[] = [];
     let total_collisions = 0;
 
-    for (const o of Array.prototype.slice.call(document.querySelectorAll("*"))) {
+    const element: NodeList = document.querySelectorAll("*");
+
+    for (const o of Array.prototype.slice.call(element)) {
         if (o.id && ids.has(o.id)) {
             const el = collisions.find((c) => c.id === o.id);
 
             ids.add(o.id);
             total_collisions += 1;
 
             if (!el) {
-                const tag = o.tagName.toLowerCase();
+                const tag: string = o.tagName.toLowerCase();
+                const id: string = o.id;
+                const className: string = o.className;
```

Comment:
The TLS and `eslint`  were not detecting the correct types so I thought it was neccesary to annotate the types manually. I can revert the changes though

Reply:
It’s always better to avoid getting a value of type `any` in the first place, than to get a value of type `any` and then annotate it with some type, because there’s nothing to guard against your annotation being wrong.

In this case, deleting `Array.prototype.slice.call` will prevent you from getting a value of type `any` in the first place.